### ISSUE TITLE
Chat feature

### ISSUE THREAD

**Issue by ericsonmichaelj** [DudeWheresMySemiColon/Legacy#26]
Chat feature won't work unless you refresh the page. Once one goes back to the main view, the drop-down menu breaks.

**Comment by sonny-qa** [comment]
i think because the app is designed without any factories, (e.g. everything is inside 'homecontroller') the socket connection doesn't persist when we move between views. Without us restructuring it, i'm wondering if there is an alternative way to solve this refresh issue

**Comment by ericsonmichaelj** [comment]
I made a few modifications to it in order to get it to work. It looked like the issue was that $scope.rooms, which was used for the menu was populated in the socket.io callback. However, $scope.rooms wasn't populated outside of its scope. 

I changed it so that the room names are already populated in the outer scope. 

EDIT:

Turns out I was wrong about the reasoning behind it. Its actually that $scope.rooms just doesn't get populated because the socket IO is already setup when you go to chats, while all the variables in home.js are refreshed. That's why refreshing it causes chat to work, because the socket IO connection gets disconnected and reconnected. 

So, yes you were right, the data for $scopes.rooms should've been stored in the factory to avoid the issue.
